test/tools/ossfuzz/SolidityEvmoneInterface.cpp

```diff
@@ -0,0 +1,197 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+
+#include <test/tools/ossfuzz/SolidityEvmoneInterface.h>
+
+#include <liblangutil/Exceptions.h>
+#include <liblangutil/SourceReferenceFormatter.h>
+
+using namespace solidity::test::fuzzer;
+
+SolidityCompilationFramework::SolidityCompilationFramework(langutil::EVMVersion _evmVersion)
+{
+	m_evmVersion = _evmVersion;
+}
+
+solidity::bytes SolidityCompilationFramework::compileContract(
+	std::string const& _sourceCode,
+	std::string const& _contractName,
+	std::map<std::string, solidity::util::h160> _libraryAddresses,
+	frontend::OptimiserSettings _optimization
+)
+{
+	std::string sourceCode = _sourceCode;
+	m_compiler.setSources({{"", sourceCode}});
+	m_compiler.setLibraries(move(_libraryAddresses));
+	m_compiler.setEVMVersion(m_evmVersion);
+	m_compiler.setOptimiserSettings(_optimization);
+	if (!m_compiler.compile())
+	{
+		langutil::SourceReferenceFormatter formatter(std::cerr, false, false);
+
+		for (auto const& error: m_compiler.errors())
+			formatter.printExceptionInformation(
+				*error,
+				formatter.formatErrorInformation(*error)
+			);
+		std::cerr << "Compiling contract failed" << std::endl;
+	}
+	evmasm::LinkerObject obj = m_compiler.object(
+		_contractName.empty() ?
+		m_compiler.lastContractName() :
+		_contractName
+	);
+	return obj.bytecode;
+}
+
+bool EvmoneUtility::isOutputExpected(
+	uint8_t const* _result,
+	size_t _length,
+	std::vector<uint8_t> const& _expectedOutput
+)
+{
+	if (_length != _expectedOutput.size())
+		return false;
+
+	return (memcmp(_result, _expectedOutput.data(), _length) == 0);
```

Comment:
```suggestion
	return ranges::span(_result, _length) == ranges::span(_expectedOutput);
```
Or given this, it may be ok to remove this function and do this inline instead :-).
If inlined, I'm also not sure if the global zeroed ``expectedOutput`` is really needed - you could just do
``_length == 32 && ranges::all_of(_expectedOutput, [](uint8_t _v) { return _v == 0; })`` or something like that :-).

Reply:
`_length == 32 && ranges::all_of(_result, [](uint8_t _v) { return _v == 0; })` does not capture the length of `uint8_t const* _result`. So, how can it say all the 32 bytes are zeroed out?

Reply:
It works if `_result` is of type `vector<uint8_t>` but `evmc` types it to be a raw mem ptr `uint8_t const*`

Reply:
I proibably wanted to write ``_length == 32 && ranges::all_of(ranges::span(_result, _length), [](uint8_t _v) { return _v == 0; })``. The first conjunct makes sure the range is 32 bytes wide, the second conjunct makes sure all of those are zero, even for "pointer plus length".

Reply:
But also fine not to go that way ;-).

Reply:
Refactored the function as suggested.